### PR TITLE
mgmt-gateway: add e2e testing

### PR DESCRIPTION
**Pull request by asm0deuz**
New mgmt-gateway (https://github.com/ceph/ceph/pull/57535) Teuthology test scenario

### CODE REVIEW

**Comment by asm0deuz** [comment]
jenkins test make check

**Comment by asm0deuz** [comment]
jenkins test make check

**Comment by asm0deuz** [comment]
https://pulpito.ceph.com/tonay-2024-10-17_17:16:21-orch:cephadm-wip-asm0deuz-testing-2024-10-17-1407-distro-default-smithi/

**Comment by asm0deuz** [comment]
Jenkins test make check

**Comment by asm0deuz** [comment]
jenkins test make check

File: qa/suites/orch/cephadm/workunits/task/test_mgmt_gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+- cephadm.shell:
+    host.a:
+      - |
+        set -ex
+        # retrieve mgmt hostname and ip
+        MGMT_GTW_HOST=$(ceph orch ps --daemon-type mgmt-gateway -f json | jq -e '.[]' | jq -r '.hostname')
+        MGMT_GTW_IP=$(ceph orch host ls -f json | jq -r --arg MGMT_GTW_HOST "$MGMT_GTW_HOST" '.[] | select(.hostname==$MGMT_GTW_HOST) | .addr')
+        # check mgmt-gateway health
+        curl -k -s https://${MGMT_GTW_IP}/health
+        curl -k -s https://${MGMT_GTW_IP}:29443/health
```
Comment (adk3798) [inline qa/suites/orch/cephadm/workunits/task/test_mgmt_gateway.yaml:68]: just curious about the difference between these two checks. Are they actually different endpoints or are we testing whether we need the port?

File: qa/suites/orch/cephadm/workunits/task/test_mgmt_gateway.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            - ${HOSTNAME}
+        spec:
+          ssl_protocols:
+            - TLSv1.2
+            - TLSv1.3
+          ssl_ciphers:
+            - AES128-SHA
+            - AES256-SHA
+          enable_health_check_endpoint: True
+        EOT
```
Comment (adk3798) [inline qa/suites/orch/cephadm/workunits/task/test_mgmt_gateway.yaml:49]: not going to require it for this PR, but we do have a `cephadm.apply` task made for doing this sort of thing. I guess we'd just have to figure out how to integrate it with your use of `${HOSTNAME}`

**Review by adk3798 — APPROVED** [review]
The test seems to be applying a mgmt-gateway spec and verifying it was deployed properly so I'm happy with it as a start. Left a couple comments but nothing I'd require for merging.

File: qa/suites/orch/cephadm/workunits/task/test_mgmt_gateway.yaml (hunk 2 above)
Comment (asm0deuz) [reply qa/suites/orch/cephadm/workunits/task/test_mgmt_gateway.yaml:49]: Would be nice to start documenting available commands in teuthology

File: qa/suites/orch/cephadm/workunits/task/test_mgmt_gateway.yaml (hunk 1 above)
Comment (asm0deuz) [reply qa/suites/orch/cephadm/workunits/task/test_mgmt_gateway.yaml:68]: This is the port for the internal server:

```server {
    listen              29443 ssl;
    listen              [::]:29443 ssl;
    ssl_certificate     /etc/nginx/ssl/nginx_internal.crt;
    ssl_certificate_key /etc/nginx/ssl/nginx_internal.key;
    ssl_protocols       TLSv1.3;
```
And it also needs to be checked

**Pull request merged**